app/models/facility_group.rb

```diff
@@ -24,8 +24,26 @@ class FacilityGroup < ApplicationRecord
   friendly_id :name, use: :slugged
 
   auto_strip_attributes :name, squish: true, upcase_first: true
+  attribute :enable_diabetes_management
+
+  before_save :enable_diabetes_management!, if: -> { enable_diabetes_management }
+  before_save :disable_diabetes_management!, if: -> { !enable_diabetes_management && diabetes_enabled? }
 
   def report_on_patients
     Patient.where(registration_facility: facilities)
   end
+
+  def diabetes_enabled?
+    facilities.where(enable_diabetes_management: false).count.zero?
+  end
+
+  private
+
+  def enable_diabetes_management!
+    facilities.update(enable_diabetes_management: true)
```

Comment:
Do any validation errors here propagate back up to the FacilityGroup? If this update fails, how does the user know it failed on the FG edit screen?

Reply:
Validation errors won't propagate from here but since we are explicitly setting it to `true` or `false`, validation shouldn't be an issue. If the update fails for other reasons it will throw an exception, which should be ok.

Reply:
Looking into wrapping this up now. Seeing if I can propagate child errors to the parent. This kind of stuff is very annoying in rails :|

Reply:
This is done:

![image](https://user-images.githubusercontent.com/50663/85138995-6720dc00-b261-11ea-8b49-d8f8ef5262f3.png)